src/btree/row_modify.c

```diff
@@ -136,6 +136,17 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
          * serialization function lock acts as our memory barrier to flush this write.
          */
         upd->next = old_upd;
+        /*
+         * Verify that we are not appending a new update in front of a prepared update that was
+         * restored from data store.
+         */
+        WT_ASSERT(session,
```

Comment:
I think we shouldn't include this assert in the change. If we really want to do this, better to add a comment to explain why we need
```
(upd->type == WT_UPDATE_TOMBSTONE && upd->txnid == WT_TXN_NONE &&
              upd->start_ts == WT_TS_NONE)
```
This is because we need to be able to use a globally visible tombstone to delete the key.

Reply:
I agree we can skip this assert as we will eventually catch this scenario anyway.